src/package-hoister.js

```diff
@@ -173,7 +176,8 @@ export default class PackageHoister {
           if (areDependenciesFulfilled) {
             // Move the package inside our sorted queue
             sortedQueue.push(queue[t]);
-            queue.splice(t--, 1);
+            // Removing items from an array in a loop is slow on large repos, but deleting is fast
+            delete queue[t];
```

Comment:
I'm not too fond of the `delete` - it basically circumvent the Flow typing, which isn't great 😕

Reply:
I believe the Flow problem can be resolved by using `queue.filter(Boolean)` below instead, [as explained here](https://github.com/facebook/flow/issues/1026).